introduction-to-dom-and-its-manipulation.html

```diff
@@ -0,0 +1,358 @@
+
+%% css
+#test {
+  padding: 1em;
+  margin: 1em;
+  width: 100px;
+  background: #bdc3c7
+}
+
+%% md
+We will now perform a very simple manipulation to the above created element using JavaScript
+
+%% js
+// Get element from the DOM tree
+a = document.getElementById('test')
+// Change a style property
+a.style.background = "#3498db"
+
+%% md
+As you would have noticed, the background color of the box has changed without the need of re-rendering that cell. DOM APIs consits of functions similar to `getElementById` such as `querySelectorAll`, `querySelector`, `getElementsByTagName` etc. which provides us access to different elements on the DOM tree.
+
+How about we make something more interactive.  
+In the next example we are going to create a collection of boxes and add hover events to them.  
+Let us write some styling code first. One for the container which we are wrapping around our boxes and the other for the box itself.
+
+%% css
+.container {
+  padding: 2em;
+}
+
+.box {
+  width: 100px;
+  height: 100px;
+  background: #8e44ad;
+  display: inline-block;
+  transition: ease 0.5s
+}
+
+%% md
+<div class="container">
+  <div class="box"></div>
+  <div class="box"></div>
+  <div class="box"></div>
+  <div class="box"></div>
+  <div class="box"></div>
+  <div class="box"></div>
+</div>
+
+%% md
+Here we have a static collection of boxes. We will now write some JavaScript event handlers to add effect on mouse hover over the box. Mouse hover can be decomposed into two components, `mouseover` and `mouseout`
+
+%% js
+var box = document.querySelectorAll(".box");
+var mouseIn = function() {
+	this.style.background = "#34495e" 
+    this.style.transform = "translateY(-10px)"
+};    
+var mouseOut = function() {	
+	this.style.background = "#8e44ad"
+	this.style.transform = "translateY(0px)"
+};
+    
+Array.from(box).forEach(function(element) {  
+	element.addEventListener('mouseover', mouseIn);
+    element.addEventListener('mouseout', mouseOut);
+});
+
+%% md
+Try hovering over the boxes to see the effect we just created.
+
+Lets carry what we have learnt to create a chart on the DOM
+
+%% md
+<div style="display: flex; align-items: flex-end">
+  <div class="bar-chart" style="height: 40px;">4</div>
+  <div class="bar-chart" style="height: 230px;">23</div>
+  <div class="bar-chart" style="height: 150px;">15</div>
+  <div class="bar-chart" style="height: 180px;">18</div>
+  <div class="bar-chart" style="height: 90px;">9</div>
+  <div class="bar-chart" style="height: 350px;">35</div>
+</div>
+<style>
+  .bar-chart {
+    font: 14px sans-serif;
+    background-color: steelblue;
+    padding: 3px 10px;
+    width: 20px;
+    margin: 1px 5px;
+    color: white
+  }
+</style>
+
+%% md
+Hmm.. what if we could automate the creation of those divs and add more interactivity to it ?
+
+Here comes D3 to our help. D3 is a JS library which helps us to interact with DOM based on data we have. We will quickly import it below.
```

Comment:
"Here comes D3 to our help." is somewhat awkward. How about something like "Luckily, there are great libraries that can help us manipulate the DOM, for example D3"